src/Umbraco.Web/Trees/ContentTreeControllerBase.cs

```diff
@@ -286,5 +301,32 @@ internal bool CanUserAccessNode(IUmbracoEntity doc, IEnumerable<MenuItem> allowe
         {
             return allowedUserOptions.Select(x => x.Action).OfType<ActionBrowse>().Any();
         }
+
+        /// <summary>
+        /// Get an entity via an id that can be either an integer or a Guid
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        internal IUmbracoEntity GetEntityFromId(string id)
+        {
+            IUmbracoEntity entity;
+            Guid idGuid = Guid.Empty;
+            int idInt;
+            if (Guid.TryParse(id, out idGuid))
+            {
+                entity = Services.EntityService.GetByKey(idGuid, UmbracoObjectType);
+
+            }
+            else if (int.TryParse(id, out idInt))
+            {
+                entity = Services.EntityService.Get(idInt, UmbracoObjectType);
+            }
+            else
+            {
+                throw new InvalidCastException("Id must be either an integer or a Guid");

```

Comment:
I'd probably just return a null here since you expecting null elsewhere when nothing is returned and never actually handling this exception